code/tests/NFluent.Tests/DateTimeRelatedTests.cs

```diff
@@ -957,6 +958,56 @@ public void AndOperatorCanChainMultipleAssertionOnDateTime()
                 .IsInstanceOf<DateTime>().And
                 .IsNotEqualTo("Batman").And.IsNotEqualTo(new DateTime(2014));
         }
+        
+        [Test]
+        public void IsCloseToTimeSpanWorks()
+        {
+            var reference = DateTime.UtcNow;
+            
+            Check.That(reference).IsCloseTo(reference.AddMilliseconds(1), TimeSpan.FromMilliseconds(1));
+            Check.That(reference).IsCloseTo(reference.AddMilliseconds(-1), TimeSpan.FromMilliseconds(1));
+        }
+        
+        [Test]
+        public void IsCloseToDurationWorks()
+        {
+            var reference = DateTime.UtcNow;
+            
+            Check.That(reference).IsCloseTo(reference.AddMilliseconds(1), new Duration(1, TimeUnit.Milliseconds));
+            Check.That(reference).IsCloseTo(reference.AddMilliseconds(-1), new Duration(1, TimeUnit.Milliseconds));
+        }
+
+        [Test]
+        public void IsCloseToTimeSpanShouldFailsIfToFar()
+        {
+            var reference = DateTime.UtcNow;
+            var expected = reference.AddMilliseconds(2);
+            var within = TimeSpan.FromMilliseconds(1);
+
+            Check.ThatCode(() => Check.That(reference).IsCloseTo(expected, within))
```

Comment:
You should use a constant date so that the error message is a constant! In this case, it would highlight that TimeSpan does not provide a simple (to read) output.

Reply:
I think that Duration is nice but I still would like to provide a TimeSpan overload. Many users might be already using extensions methods like FluentDateTime to create TimeSpan.